.github/workflows/charm-release.yaml

```
@@ -78,10 +78,16 @@ jobs:
           sudo snap install charmcraft --classic --channel latest/stable
           cd $GITHUB_WORKSPACE/charm/${{ inputs.charm-path }}
           # Get the libs folder name
-          libraries_folder=$(yq .name metadata.yaml | tr - _)
-          # For each library belonging to the charm, publish it
-          for lib in $(find lib/charms/$libraries_folder -type f | sed 's|lib/||' | sed 's/.py//' | sed 's|/|.|g'); do
-            charmcraft publish-lib $lib
-          done
+          if [ -e metadata.yaml ]; then
+            libraries_folder=$(yq .name metadata.yaml | tr - _)
+          else
+            libraries_folder=$(yq .name charmcraft.yaml | tr - _)
+          fi
+          if [ -e lib/charms/$libraries_folder ]; then
+            # For each library belonging to the charm, publish it
+            for lib in $(find lib/charms/$libraries_folder -type f | sed 's|lib/||' | sed 's/.py//' | sed 's|/|.|g'); do
+              charmcraft publish-lib $lib
+            done
+          fi
```

This `if` isn't necessary, it just adds a level of indentation; `find` won't return anything if there's no folder, meaning the `for` loop won't be executed; I feel like adding nesting makes things less readable :)

```suggestion
          # For each library belonging to the charm, publish it
          for lib in $(find lib/charms/$libraries_folder -type f | sed 's|lib/||' | sed 's/.py//' | sed 's|/|.|g'); do
            charmcraft publish-lib $lib
          done
```

If we **really** think we should keep it we should at least replace `-e` with `-d`, but I'd rather remove this `if` entirely :D